src/plots/VolumeVTK9/avtVolumeFilter.C

```diff
@@ -0,0 +1,1462 @@
+    #include <avtSLIVRRayTracer.h>
+#endif
+
+//
+// Function Prototypes
+//
+
+static void CreateViewInfoFromViewAttributes(avtViewInfo &,
+                                             const View3DAttributes &);
+
+
+// ****************************************************************************
+//  Method: avtVolumeFilter constructor
+//
+//  Programmer: Hank Childs
+//  Creation:   November 20, 2001
+//
+//  Modifications:
+//
+//    Hank Childs, Wed Nov 24 16:23:41 PST 2004
+//    Removed references to data members that have been moved.
+//
+// ****************************************************************************
+
+avtVolumeFilter::avtVolumeFilter()
+{
+}
+
+
+// ****************************************************************************
+//  Method: avtVolumeFilter destructor
+//
+//  Programmer: Hank Childs
+//  Creation:   November 20, 2001
+//
+//  Modifications:
+//
+//    Hank Childs, Wed Nov 24 16:23:41 PST 2004
+//    Removed references to data members that have been moved.
+//
+// ****************************************************************************
+
+avtVolumeFilter::~avtVolumeFilter()
+{
+    if (primaryVariable != nullptr)
+    {
+        delete [] primaryVariable;
+        primaryVariable = nullptr;
+    }
+
+    if (VisItVTKRenderFilter != nullptr)
+    {
+        delete VisItVTKRenderFilter;
+        VisItVTKRenderFilter = nullptr;
+    }
+}
+
+
+// ****************************************************************************
+//  Method: avtVolumeFilter::SetAttributes
+//
+//  Purpose:
+//      Sets the attributes of the software override filter.
+//
+//  Arguments:
+//      a       The attributes for the filter.
+//
+//  Programmer: Hank Childs
+//  Creation:   November 20, 2001
+//
+// ****************************************************************************
+
+void
+avtVolumeFilter::SetAttributes(const VolumeAttributes &a)
+{
+    atts = a;
+}
+
+
+// ****************************************************************************
+//  Method: avtVolumeFilter::Execute
+//
+//  Purpose:
+//      Just passes the data through.  The volume filter really only does
+//      the work in "RenderImage".
+//
+//  Programmer: Hank Childs
+//  Creation:   November 11, 2004
+//
+//  Modifications:
+//
+//    Hank Childs, Fri May 21 11:28:12 CDT 2010
+//    Calculate histograms for setting up transfer functions.
+//
+//    Hank Childs, Wed Feb  2 17:51:34 CST 2011
+//    Add timings statement for histogram calculation.
+//
+//    Kathleen Biagas, Wed Nov 18 2020
+//    Replace VISIT_LONG_LONG with long long.
+//
+// ****************************************************************************
+
+void
+avtVolumeFilter::Execute(void)
+{
+    // Copy our input to the output
+    avtDataObject_p input = GetInput();
+    GetOutput()->Copy(*input);
+
+    avtDataset_p ds = GetTypedInput();
+
+    // The rest of this method is to set up histogram.
+    int numValsInHist = 256;
+    double minmax[2] = { 0, 1 };
```

Comment:
Is initialization needed here given 172 and 173 in same scope. Reason I ask is that `0,1` a bit odd for initialization values. I would have expected either `0,0` or maybe `DBL_MAX,-DBL_MAX`.

Reply:
Probably to squash a compiler warning about uninitialized var being sent to a function (line 165).